Code/EditorPlugins/Assets/EditorPluginAssets/CustomDataAsset/CustomDataAsset.cpp

```diff
@@ -52,3 +52,29 @@ ezTransformStatus ezCustomDataAssetDocument::InternalTransformAsset(ezStreamWrit
   ezAbstractGraphBinarySerializer::Write(stream, &abstractObjectGraph);
   return ezStatus(EZ_SUCCESS);
 }
+
+void ezCustomDataAssetDocument::UpdateAssetDocumentInfo(ezAssetDocumentInfo* pInfo) const
+{
+  SUPER::UpdateAssetDocumentInfo(pInfo);
+
+  const ezDocumentObject* pObject = GetPropertyObject();
+
+  ezVariant type = pObject->GetTypeAccessor().GetValue("Type");
+  EZ_ASSERT_DEV(type.IsA<ezUuid>(), "Implementation error");
+
+  if (const ezDocumentObject* pDataObject = pObject->GetChild(type.Get<ezUuid>()))
+  {
```

Comment:
Try not to use `GetTypeAccessor` on the object or `GetCild` as these functions are implementation details and require various assumptions. Better use the `ObjectAccessor`:

```suggestion
  const ezUuid typeGuid = GetObjectAccessor()->Get<ezUuid>(pObject, "Type");
  if (const ezDocumentObject* pDataObject = GetObjectAccessor()->GetObject(typeGuid))
  {
```